platform/event/event_descriptor.proto

```diff
@@ -0,0 +1,64 @@
+/**
+ * # Event Descriptor
+ * Unique identifier for an event.
+ *
+ * ### Keywords
+ * The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+ * "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+ * document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119).
+ */
+syntax = "proto3";
+
+package com.hedera.hapi.platform.event;
+
+/*
+ * Hedera Network Services Protobuf
+ *
+ * Copyright (C) 2018 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+option java_package = "com.hedera.hapi.platform.event";
+// <<<pbj.java_package = "com.hedera.hapi.platform.event">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * Unique identifier for an event.
+ */
+message EventDescriptor {
+  /**
+   * The hash of the event. <br/>
+   * Future work: the precise hash methodology hasn't been determined yet.
+   */
+  bytes hash = 1;
+
+  /**
+   * The ID of the node that created the event.
+   */
+  int64 creator_id = 2;
+
+  /**
+   * The generation of the event. <br/>
+   * The generation SHALL be equal to the maximum generation of all parents + 1. <br/>
```

Comment:
Could be a little clearer (and the space before the break is unnecessary)
```suggestion
   * A consensus event generation.<br/>
   * This value SHALL be **one greater** than the _maximum_ generation of all parents.<br/>
```

Reply:
This is why we're working to improve the specification, to make these things clear.
Might be worth noting in the event descriptor message comment that this may be used for both consensus and non-consensus events (do we have specific types on non-consensus events?).

Reply:
The latest plan is to *not* have separate types for consensus / non-consensus events. 

I'll tweak the event descriptor message comment, to make it clear that the event descriptor can be constructed immediately, without needing to wait for the event to reach consensus